packages/api/src/lib/granule-delete.ts

```diff
@@ -97,7 +99,7 @@ const deleteGranuleAndFiles = async (params: {
     logger.debug(`Successfully removed S3 files ${JSON.stringify(apiGranule.files)}`);
     return;
   }
-  if (pgGranule.published) {
+  if (pgGranule && pgGranule.published) {
```

Comment:
This file's modifications are *weird*.        master has updates to this file that aren't part of the merge.    Bad prior merge being noted as sorted via git?

Reply:
Oh this one I can at least partially explain. I added the `pgGranule`. Previously we were doing a separate check for something along the lines of `if (pgGranule) ...` above the published check. That was removed but it seemed to me that we should still validate `pgGranule` before assuming it has a property.

Reply:
In TS should we just use optional chaining, probably with an explicit assertion?

e.g.
 
```
if(pgGranule.?published === true) {

}
```

Publish can't actually be set to '1' or something, but it's probably still reasonable to be explicit, *particularly* given all of the upstream source may not be TS yet.

Reply:
No, that's just an oops on my part while transcribing to github.   👍🏻